### PR TITLE
add on_missing keyword to mne.chpi.get_active_hpi()

### LINKED ISSUE #11162: getting chpi information crashes when no hpi information are available
### Describe the new feature or enhancement

The new ( #11122 ) method `mne.chpi.get_active_chpi` calls `mne.chpi. get_chpi_info` to determine the number of active hpi coils. In case no information on hpis are available, the default behavior of `get_chpi_info` is raising an error. This is not ideal for all situations (see for example  [mne-bids #1067](https://github.com/mne-tools/mne-bids/pull/1067). Therefore it would be nice to be able to modify that behavior. 

### Describe your proposed implementation

I would say adding the `on_missing` keyword to  `mne.chpi.get_active_chpi` and passing it on to `mne.chpi. get_chpi_info` should be easy and sufficient. 

### Describe possible alternatives

alternatively, one could set the default behavior of the on_missing field to something less restrictive than raise. Not sure whether that would cause problems elsewhere.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by eort**
#### What does this implement/fix?
Fixes #11162 by adding the `on_missing` argument to the get_active_hpi function

#### Additional information
Not sure why, but I have problems using the `%(on_missing_chpi)s` pattern, even though I just copy&pasted within that same file. So the documentation doesn't properly render yet.

### CODE REVIEW

**Comment by drammock** [comment]
> Not sure why, but I have problems using the `%(on_missing_chpi)s` pattern,

make sure the function has a `@fill_doc` decorator (or `@verbose` will also work, but only use that if the function has a `verbose` param)

**Comment by eort** [comment]
> make sure the function has a `@fill_doc` decorator

thanks! Good to know

File: mne/chpi.py [hunk 1]
```diff
@@ -1403,6 +1406,9 @@ def get_active_chpi(raw):
     -----
     .. versionadded:: 1.2
     """
+    _check_option(parameter='on_missing', value=on_missing,
+                  allowed_values=['ignore', 'raise', 'warn'])
+
```
Comment (larsoner) [inline mne/chpi.py:1411]: You don't need this here as it's handled already in `get_chpi_info`
```suggestion
```

File: mne/chpi.py [hunk 2]
```diff
@@ -48,6 +48,7 @@
 from .utils import (verbose, logger, use_log_level, _check_fname, warn,
                     _validate_type, ProgressBar, _check_option, _pl,
                     _on_missing)
+from .utils.docs import fill_doc
```
Comment (larsoner) [inline mne/chpi.py:51]:
```suggestion
                    _on_missing, verbose)
```

File: mne/chpi.py [hunk 3]
```diff
@@ -1386,13 +1387,15 @@ def _compute_good_distances(hpi_coil_dists, new_pos, dist_limit=0.005):
     return use_mask, these_dists
 
 
-def get_active_chpi(raw):
+@fill_doc
+def get_active_chpi(raw, on_missing='raise'):
```
Comment (larsoner) [inline mne/chpi.py:1391]:
```suggestion
@verbose
def get_active_chpi(raw, *, on_missing='raise', verbose=None):
```

File: mne/chpi.py [hunk 4]
```diff
@@ -1386,13 +1387,15 @@ def _compute_good_distances(hpi_coil_dists, new_pos, dist_limit=0.005):
...
     """Determine how many HPI coils were active for a time point.
 
     Parameters
     ----------
     raw : instance of Raw
         Raw data with cHPI information.
+    %(on_missing_chpi)s
```
Comment (larsoner) [inline mne/chpi.py:1398]:
```suggestion
    %(on_missing_chpi)s
    %(verbose)s
```

**Review by larsoner — COMMENTED** [review]
+1 for going to `verbose`, I think my suggestions should do it

**Review by agramfort — APPROVED** [review]

**Comment by larsoner** [comment]
Thanks @eort !

**Pull request merged**